### PR TITLE
Fix Router to handle same url with different HTTP methods

### PR DESCRIPTION
**Pull request by joanromano**
Might be also of interest for @leviathan

### CODE REVIEW

File: Tests/RouterTests.swift [hunk 1]
```diff
@@ -213,6 +213,70 @@ class RouterTests: QuickSpec {
...
+                expect(calledPut).toEventually(beTrue())
+                
+                request = NSMutableURLRequest(URL: NSURL(string: "http://www.test.com/users/1")!)
+                request.HTTPMethod = "DELETE"
+                NSURLSession.sharedSession().dataTaskWithRequest(request) { (_, _, _) in }.resume()
+                
+                expect(calledDel).toEventually(beTrue())
+            }
+            
+            it("should properly replace handlers with same registered url and http methods") {
```
Comment (MP0w) [inline Tests/RouterTests.swift]: properly can actually be avoided, no?

File: Tests/RouterTests.swift [hunk 2]
```diff
@@ -213,6 +213,70 @@ class RouterTests: QuickSpec {
                 expect(usersCommentsResponseURL?.absoluteString).toEventually(equal("http://www.test.com/users/1/comments/2?page=2&author=hector"))
             }
             
+            it("should call handlers with same registered url but different http methods") {
+                var calledPost = false
+                var calledPut = false
+                var calledDel = false
```
Comment (MP0w) [inline Tests/RouterTests.swift:219]: just wondering, why did you also add `DELETE`?

File: Tests/RouterTests.swift [hunk 3]
```diff
@@ -213,6 +213,70 @@ class RouterTests: QuickSpec {
                 expect(usersCommentsResponseURL?.absoluteString).toEventually(equal("http://www.test.com/users/1/comments/2?page=2&author=hector"))
             }
             
+            it("should call handlers with same registered url but different http methods") {
```
Comment (MP0w) [inline Tests/RouterTests.swift]: for the 2 new tests what about "...with the same path..."?

**Review by MP0w — APPROVED** [review]

File: Tests/RouterTests.swift (hunk 2 above)
Comment (joanromano) [reply Tests/RouterTests.swift:219]: For the lulz driven development

**Pull request merged**